clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -4058,6 +4058,33 @@ class CIRThrowOpLowering
   }
 };
 
+class CIRIsFPClassOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::IsFPClassOp> {
+public:
+  using OpConversionPattern<mlir::cir::IsFPClassOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::IsFPClassOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto src = adaptor.getSrc();
+    auto flags = adaptor.getFlags();
+    auto retTy = rewriter.getI1Type();
+
+    auto loc = op->getLoc();
+
+    auto intrinsic = createCallLLVMIntrinsicOp(
+        rewriter, loc, "llvm.is.fpclass", retTy, mlir::ValueRange{src, flags});
```

Comment:
Instead of `llvm.call_intrinsic`, use `llvm.intr.is.fpclass` instead:

```suggestion
    auto intrinsic = rewriter.create<mlir::LLVM::IsFPClass>(/* ... */);
```

Reply:
Done. I meant to use the consistent API but I can't use it after I make the flags to be an Attr.